docs/getting-started/src/getting-started/counter/deploy-and-solve.md

```diff
@@ -0,0 +1,172 @@
+predicate.
+
+A solution requires 3 components:
+
+1. **`predicate_to_solve`**: The full address of the predicate we wish to
+   solve. This includes both the contract and predicate content addresses. If
+   we look back at our `pint build` step, we'll notice that these addresses are
+   printed to the command
+   line:
+   ```console
+   $ pint build
+      Compiling counter [contract] (/Users/mindtree/Desktop/counter/contract)
+       Finished build [debug] in 13.38825ms
+       contract counter            1899743AA94972DDD137D039C2E670ADA63969ABF93191FA1A4506304D4033A2
+            └── counter::Increment 355A12DCB600C302FFD5D69C4B7B79E60BA3C72DDA553B7D43F4C36CB7CC0948
+   ```
+   Note that your addresses might be slightly different if using a newer
+   compiler version.
+
+2. **`decision_variables`**: A list of parameters (aka decision variables) that
+   are expected by the predicate. Our `Increment` predicate takes no
+   parameters, so here we can specify an empty list.
+
+3. **`state_mutations`**: The state mutations we wish to propose. In our case,
+   we want to initialise the `counter` storage variable to `1`.
+
+   > **Note:** In state mutations, both keys and values are variable-sized
+   > arrays of `Word`s. As `counter` is the only variable, we can assume it is
+   > at key `[0]`. As the type of `counter` is `int`, we only need a single
+   > `Word` to represent the value, e.g. `[1]`.
+
+When represented as JSON, our full solution looks as follows:
+
+```json
+{
+    "data": [
+        {
+            "predicate_to_solve": {
+                "contract": "1899743AA94972DDD137D039C2E670ADA63969ABF93191FA1A4506304D4033A2",
+                "predicate": "355A12DCB600C302FFD5D69C4B7B79E60BA3C72DDA553B7D43F4C36CB7CC0948"
+            },
+            "decision_variables": [],
+            "state_mutations": [
+                {
+                    "key": [0],
+                    "value": [1]
+                }
+            ]
+        }
+    ]
+}
+```
+
+Lets put the above JSON in a `solution.json` file.
+
+To submit our solution to the local builder, we can now use the following command:
+
+```
+essential-rest-client submit-solution "http://127.0.0.1:57968" ./solution.json
+```
+
+As confirmation that the builder received our solution, it responds with its
+content address.
+
+However, this is not enough to know whether or not our solution was included in
+a block, or whether it passed the contract's constraints at all.
+
+To check whether or not our solution was successful, we can query the state of
+our contract's `counter` storage variable using the builder's node API:
+
+```console
+$ essential-rest-client query-state --content-address "1899743AA94972DDD137D039C2E670ADA63969ABF93191FA1A4506304D4033A2" "http://127.0.0.1:59438" "0000000000000000"
+```
+
+Here, we're providing the counter's contract address:
+
+```
+--content-address "1899743AA94972DDD137D039C2E670ADA63969ABF93191FA1A4506304D4033A2"
+```
+
```

Comment:
We should probably also mention that we have to provide the node server API address